sound/soc/sof/sof-audio.h

```diff
@@ -109,6 +111,8 @@ struct sof_ipc_tplg_control_ops {
  * @token_list: List of token ID's that should be parsed for the widget
  * @token_list_size: number of elements in token_list
  * @bind_event: Function pointer for binding events to the widget
+ * @prepare: Optional op for preparing a widget for set up
+ * @unprepare: Optional op for unpreparing a widget
```

Comment:
can prepare be omitted and unprepare present? or is unprepare only possible when a prepare is provided?

Reply:
you can have unprepare without a prepare but without a prepare, a widget's complete flag will never be set, so unprepare() would never ever get invoked.